src/main/java/com/synopsys/integration/detect/lifecycle/run/step/BinaryScanStepRunner.java

```diff
@@ -70,4 +72,16 @@ public Optional<File> determineBinaryScanFileTarget(DockerTargetData dockerTarge
             return Optional.empty();
         }
     }
+    
+    public Optional<String> extractBinaryScanId(UploadFinishResponse response) {     
+        try {
+            String location = response.getLocation();
+            URI uri = new URI(location);
+            String path = uri.getPath();
+            String scanId = path.substring(path.lastIndexOf('/') + 1);
+            return Optional.of(scanId);
+        } catch (Exception e) {
+            return Optional.empty();
```

Comment:
It might be good to log an error so we're not completely silencing an exception.